packages/botonic-plugin-flow-builder/src/index.ts

```diff
@@ -91,11 +93,30 @@ export default class BotonicPluginFlowBuilder implements Plugin {
       request.input.payload = this.cmsApi.getPayload(nodeByUserInput?.target)
     }
 
+    this.updateRequestBeforeRoutes(request)
+  }
+
+  private updateRequestBeforeRoutes(request: PluginPreRequest) {
     if (request.input.payload) {
-      request.input.payload = request.input.payload?.split(
-        SOURCE_INFO_SEPARATOR
-      )[0]
+      request.input.payload = this.removeSourceSeparatorFromPayload(
+        request.input.payload
+      )
+      request.input.payload = this.updateBotActionPayload(request.input.payload)
+    }
+  }
+
+  private removeSourceSeparatorFromPayload(payload: string): string {
+    return payload?.split(SOURCE_INFO_SEPARATOR)[0]
+  }
+
+  private updateBotActionPayload(payload: string): string {
+    if (payload.startsWith(BOT_ACTION_PAYLOAD_SEPARATOR)) {
+      const botActionId = payload.split(SEPARATOR)[1]
+      const botActionNode =
+        this.cmsApi.getNodeById<HtBotActionNode>(botActionId)
+      return this.cmsApi.createPayloadWithParams(botActionNode)
     }
+    return payload
```

Comment:
Suggestion: I would move the condition that checks if the payload starts with the `BOT_ACTION_PAYLOAD_SEPARATOR` outside the function so the function will be executed only if we detect a bot action payload and I think the code will be clearer.